src/Compilers/CSharp/Portable/Binder/Binder_Await.cs

```diff
@@ -431,5 +432,102 @@ private static bool HasOptionalOrVariableParameters(MethodSymbol method)
 
             return false;
         }
+
+        /// <summary>
+        /// Returns the appropriate AsyncMethodBuilder based on whether the
+        /// return type is void or a non-generic or generic task-like type. The
+        /// caller is responsible for ensuring this is used for async methods only.
+        /// </summary>
+        internal AsyncMethodBuilder GetAsyncMethodBuilder(CSharpSyntaxNode syntax, TypeSymbol returnType)
```

Comment:
The method is defined on `Binder` to allow using `Binder.LookupMembersInType` to find the `CreateAsyncMethodBuilder` method, in particular to use the accessibility checking of `LookupMembersInType`. It should probably not be grouped with the partial class for Awaiter though - I'll move it.